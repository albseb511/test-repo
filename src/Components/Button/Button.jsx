import React from "react";

const Button = ({ label, color, handleClick }) => {
  const style = { backgroundColor: color };
  return (
    <div style={style} data-testid="btn-basic" onClick={handleClick}>
      {label}
    </div>
  );
};

export default Button;
