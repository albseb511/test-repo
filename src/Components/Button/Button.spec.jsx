import React from "react";
import ReactDOM from "react-dom";
import Button from "./Button";
import { render, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";

describe("Button should work correctly", () => {
  it("should render without crashing", () => {
    // const container = document.createElement("div");
    // ReactDOM.render( <Button label="button" /> ,container)
    render(<Button label="button" />);
  });

  it("should render the label correctly", () => {
    const label = "LABEL";
    const { getByTestId } = render(<Button label={label} />);
    const btn = getByTestId("btn-basic");
    expect(btn.textContent).toBe(label);
  });

  it("should render the correct color", () => {
    const label = "ADD";
    const color = "green";
    const { getByTestId, debug } = render(
      <Button label={label} color={color} />
    );
    const btn = getByTestId("btn-basic");
    expect(btn.textContent).toBe(label);
    expect(btn.style.backgroundColor).toBe(color);
  });

  it("should invoke the function callback", () => {
    const mockFn = jest.fn();
    const label = "REMOVE";

    const response = render(<Button label={label} handleClick={mockFn} />);
    const { getByTestId } = response;
    const btn = getByTestId("btn-basic");
    fireEvent.click(btn);
    expect(mockFn).toHaveBeenCalledTimes(1);
    fireEvent.click(btn);
    fireEvent.click(btn);
    expect(mockFn).toHaveBeenCalledTimes(3);
  });

  it("match the snapshot", () => {
    const tree = renderer.create(
      <Button label="REMOVE" color="red" handleClick={() => {}} />
    );
    expect(tree).toMatchSnapshot();
  });

  it("match the second snapshot", () => {
    const tree = renderer.create(
      <Button label="ADD" color="red" handleClick={() => {}} />
    );
    expect(tree).toMatchSnapshot();
  });
});
