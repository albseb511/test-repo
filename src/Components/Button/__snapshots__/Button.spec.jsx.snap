// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should work correctly match the second snapshot 1`] = `
<div
  data-testid="btn-basic"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  ADD
</div>
`;

exports[`Button should work correctly match the snapshot 1`] = `
<div
  data-testid="btn-basic"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  REMOVE
</div>
`;
